### PR TITLE
Persistent IEDs + Starting IED Threat Level

### PR DESCRIPTION
**Pull request by trapw0w**
#678 
- Added the ability to set a starting level IED threat when IED module is synced with the Military Commander. 
- Works similar to how the module operates without being synced to the Commander on initial startup.
- Military commander will continue to place IEDs as the Insurgency progresses 

#351 
- Ability to persist IED locations & triggers.
- Will persist the type of IED, e.g. Dud.
- Works in conjunction with the starting IED threat level, can also be used without it.

### Testing:
**Local DB**
- Synced, Startup IEDs, Persistence: Working **05/07/2020 AEST**
- Synced, Startup IEDs, No Persistence: Working **05/07/2020 AEST**
- Synced, No Startup IEDs, No Persistence: Working **05/07/2020 AEST**
- Synced, No Startup IEDs, Persistence: Working  **05/07/2020 AEST**
- Unsynced, Startup IEDs, Persistence: Working  **05/07/2020 AEST**
- Unsynced, Startup IEDs, No Persistence: Working  **05/07/2020 AEST**
- Unsynced, No Startup IEDs, Persistence: Working  **05/07/2020 AEST**
- Unsynced, No Startup IEDs, No Persistence: **05/07/2020 AEST**

**Cloud DB**
- Synced, Startup IEDs, Persistence (ID22): Working **05/07/2020 AEST**
- Synced, Startup IEDs, No Persistence (ID23): Working **05/07/2020 AEST**
- Synced, No Startup IEDs, No Persistence (ID24): Working **05/07/2020 AEST**
- Synced, No Startup IEDs, Persistence (ID125): Working **05/07/2020 AEST**
- Unsynced, Startup IEDs, Persistence (ID26): Working **05/07/2020 AEST**
- Unsynced, Startup IEDs, No Persistence (ID27): Working **05/07/2020 AEST**
- Unsynced, No Startup IEDs, Persistence (ID28): Working **05/07/2020 AEST**
- Unsynced, No Startup IEDs, No Persistence (ID29): Working **05/07/2020 AEST**

**All working as of:** 736cd738d83ba43bd10480edf92553c5bdaa5ff7

### Known Issues (To Fix)
=====================================
**Issue:** VBIEDs and SBIEDs won't spawn on second load (Resolved in f3c7d62de70396dd43f997ebeca9767b062893c3)
**Parameters**
  - Unsynced with Military Commander
  - Persistence Enabled

**Root Cause:**
- Location store is used to save VBIED's and SBIED's in runtime until they're made persistent at a later date. Location store is only loaded once (first run with persistence enabled) & is reset prior to a save. As a result, location store is empty on second load which results in no VBIED's or SBIED's being placed.

=====================================

### CODE REVIEW

File: addons/mil_opcom/fnc_INS_helpers.sqf [hunk 1]
```diff
@@ -270,11 +270,13 @@ ALiVE_fnc_INS_ied = {
...
                     _trg setTriggerArea [_size + 250, _size + 250,0,false];
                     _trg setTriggerActivation ["ANY","PRESENT",true];
+                    _num = ceil(_size/100);
                     _trg setTriggerStatements [
                         "this && {(vehicle _x in thisList) && ((getposATL _x) select 2 < 25)} count ([] call BIS_fnc_listPlayers) > 0",
-                            format["null = [getpos thisTrigger,%1,%2,%3] call ALIVE_fnc_createIED",_size,str(_id),ceil(_size/100)],
+                            format["null = [getpos thisTrigger,%1,%2,%3] call ALIVE_fnc_createIED",_size,str(_id),_num],
                             format["null = [getpos thisTrigger,%1] call ALIVE_fnc_removeIED",str(_id)]
                     ];
+                    ["storeTrigger", [_size,_id,_pos,_num]] call ALiVE_fnc_IED;
```
Comment (trapw0w) [inline addons/mil_opcom/fnc_INS_helpers.sqf:279]: Need to add a check above to confirm if persistence has been set on `ALiVE_MIL_IED`

File: addons/mil_ied/fnc_IED.sqf [hunk 2]
```diff
@@ -74,9 +79,8 @@ private ["_logic","_operation","_args","_result"];
 
 TRACE_1("IED - input",_this);
 
-_logic = [_this, 0, objNull, [objNull]] call BIS_fnc_param;
-_operation = [_this, 1, "", [""]] call BIS_fnc_param;
-_args = [_this, 2, objNull, [objNull,[],"",0,true,false]] call BIS_fnc_param;
+params [["_logic", objNull, [objNull]], ["_operation", "", [""]], ["_args", objNull, [objNull,[],"",0,true,false]]];
```
Comment (trapw0w) [inline addons/mil_ied/fnc_IED.sqf:82]: As per @DavisBrown723, need to convert this to multiline:

```sqf
params [
    ["_logic", objNull, [objNull]],
    ["_operation", "", [""]],
    ["_args", objNull, [objNull,[],"",0,true,false]]
];```

**Comment by svencc** [comment]
Thank you so much for implementing this feature!

**Comment by trapw0w** [comment]
Quick test to confirm IED Placement via the Military Commander:

```sqf
_objectives = [OPCOM_INSTANCES select 0, "objectives"] call ALiVE_fnc_hashGet;

{

	_objective = _x;
	_center = [_objective,"center"] call ALiVE_fnc_HashGet;
	_id = [_objective,"objectiveID",""] call ALiVE_fnc_HashGet;
	_size = [_objective,"size",-1] call ALiVE_fnc_HashGet;

	_factions = ["OPF_F"];
	_faction = selectRandom _factions;
	_roads = _center nearRoads _size;
	_target = selectRandom _roads;
	_target = [[],"convertObject",_target] call ALiVE_fnc_OPCOM;
	_sidesEnemy = ["WEST"];


	_sector = [ALIVE_sectorGrid, "positionToSector", _center] call ALIVE_fnc_sectorGrid;
	_sectorData = [_sector,"data",["",[],[],nil]] call ALIVE_fnc_hashGet;
	_civClusters = [_sectorData,"clustersCiv"] call ALIVE_fnc_hashGet;
	_settlementClusters = [_civClusters,"settlement",[]] call ALIVE_fnc_hashGet;
	_agentClusterData = [ALIVE_agentHandler,"agentsByCluster",["",[],[],nil]] call ALiVE_fnc_hashGet;
	_agents =  ([_agentClusterData,_settlementClusters select 0 select 1,["",[],[],nil]] call ALiVE_fnc_HashGet) select 1; 



	[time,_center,_id,_size,_faction,_target,_sidesEnemy,_agents] spawn ALiVE_fnc_INS_ied;

}forEach _objectives;
```

**Comment by trapw0w** [comment]
@tupolov added `mil_ied` DB to Couch DB. Will test to confirm.

**Comment by DarrellJMartin** [comment]
Resolved conflicts.

**Pull request merged**